src/Microsoft.Framework.Runtime.Roslyn/AfterCompileContext.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.Framework.Runtime.Roslyn
+{
+    internal class AfterCompileContext : IAfterCompileContext
+    {
+        private readonly CompilationContext _context;
+
+        public AfterCompileContext(CompilationContext context, IEnumerable<Diagnostic> emitDiagnostics)
+        {
+            _context = context;
+            var diagnostics = new List<Diagnostic>(context.Diagnostics);
+            diagnostics.AddRange(emitDiagnostics);
+
+            Diagnostics = diagnostics;
+        }
+
+        public IProjectContext ProjectContext
+        {
+            get
+            {
+                return _context.ProjectContext;
+            }
+        }
+
+        public Stream AssemblyStream { get; set; }
```

Comment:
Any chance we get an assembly here instead? If the AfterCompileContext doesn't change it, you'll no-op. I could use this for Razor precompilation - but with the current approach, we'll need to emit the result twice - once with this stream to find interesting types and again when we hydrate it with resources.